sql/src/stream.cc

```diff
@@ -124,6 +124,60 @@ void stream::_cache_create() {
   return ;
 }
 
+/**
+ * Create the cache to link host ids to instance ids.
+ */
+void stream::_cache_host_instance_create() {
+  _cache_host_instance.clear();
+  std::ostringstream oss;
+  oss << "SELECT host_id, instance_id"
+      << " FROM " << ((_db.schema_version() == database::v2)
+                      ? "hosts"
+                      : "rt_hosts");
+  try {
+    database_query q(_db);
+    q.run_query(oss.str());
+    while (q.next())
+      _cache_host_instance.insert(
+        std::make_pair(q.value(0).toUInt(), q.value(1).toUInt()));
+  }
+  catch (std::exception const& e) {
+    logging::error(logging::high)
+      << "SQL: could not get list of hosts/instances relations: "
+      << e.what();
+  }
+}
+
+void stream::_cache_host_instance_clean(unsigned int instance_id) {
+  for (std::map<unsigned int, unsigned int>::iterator
```

Comment:
You should use _std::multimap<unsigned int, unsigned int>_, associated the other way around (_instance_id_, _host_id_) to search inside the container instead of iterating over it.

Reply:
I know that but I need a such cache for a change I will do in a near future. But OK, I will change it later or we will add another cache...